lib/lz4.h

```diff
@@ -238,11 +238,25 @@ typedef union LZ4_stream_u LZ4_stream_t;  /* incomplete type (defined later) */
 LZ4LIB_API LZ4_stream_t* LZ4_createStream(void);
 LZ4LIB_API int           LZ4_freeStream (LZ4_stream_t* streamPtr);
 
-/*! LZ4_resetStream() :
- *  An LZ4_stream_t structure can be allocated once and re-used multiple times.
- *  Use this function to start compressing a new stream.
+/*! LZ4_resetStream_fast() : v1.9.0+
+ *  Use this to prepare an LZ4_stream_t for a new chain of dependent blocks
+ *  (e.g., LZ4_compress_fast_continue()).
+ *
+ *  An LZ4_stream_t must be initialized once.
+ *  This is automatically done when created by LZ4_createStream().
+ *  However, should the LZ4_stream_t be simply declared on stack (for example),
+ *  it's necessary to initialize first using LZ4_initStream().
+ *
+ *  After that, start any new stream with LZ4_resetStream_fast().
+ *  A same LZ4_stream_t can be re-used multiple times consecutively
+ *  and compress multiple streams,
+ *  provided that it starts each new stream with LZ4_resetStream_fast().
+ *
+ *  LZ4_resetStream_fast() is much faster than LZ4_initStream(),
+ *  but is not compatible with memory regions containing garbage data.
+ *  For this reason, LZ4_stream_t must be initialized at least once,
```

Comment:
The one topic I think this comment doesn't cover is how this interacts with the `_extState` functions. One would like to think that the interaction is trivial, and that a sequence of calls:

1. `LZ4_initStream()`
2. `LZ4_compress_fast_extState_fastReset()`
3. `LZ4_resetStream_fast()`
4. `LZ4_compress_fast_extState_fastReset()`

would be just as fast as:

1. `LZ4_initStream()`
2. `LZ4_compress_fast_extState_fastReset()`
3. `LZ4_compress_fast_extState_fastReset()`

However, they actually have different behaviors. `LZ4_resetStream_fast()` forces the stream into `byU32` mode. Whereas `LZ4_compress_fast_extState_fastReset()` on small values will use `byU16` mode. So in the first case, two (or actually is it three?) full resets happen, while in the second only one does.

In short: you shouldn't call `LZ4_resetStream_fast()` with the `*extState*` functions. Do we want to point that out? I would propose adding:

> Note: it's only useful to call this in the context of streaming compression. The `*extState*` functions perform their own resets and using this before calling them is redundant or even counterproductive.

Speaking of which, do we want to promote `LZ4_compress_fast_extState_fastReset()` to stable as well?

Reply:
Indeed, I haven't yet covered `_extState` functions.
It's good you bring that up.

The additional comment you propose looks good to me.
I'll add it.

> LZ4_compress_fast_extState_fastReset()

I'm not sure.
I was afraid of the increased complexity.

If it's just promoted, with no other change, now there will be 2 functions with a minor performance behavior. Hence a risk of confusion. And for what gain ?

I understand the gain in the context of Managed Compression,
but for external users, they may as well be lost in the subtleties.

Another solution is to go "all in", with `LZ4_compress_fast_extState_fastReset()` taking the place of `LZ4_compress_fast_extState()`.
Since `LZ4_compress_fast_extState_fastReset()` cannot completely replace `LZ4_compress_fast_extState()`, due to issue with uninitialized states, `LZ4_compress_fast_extState()` must remain available. But it can be de-prioritized.

Then it's a matter of documenting that the state must be initialized first, using `LZ4_initStream()`, before invoking `LZ4_compress_fast_extState_fastReset()`.
Also, this function name is really long and feels complex.
Not too big of an issue in the experimental section, but something to think about on reaching "stable" API status.

Reply:
_edit_ : suggested comment added

Reply:
Yeah. Happy to leave `LZ4_compress_fast_extState_fastReset()` in experimental.